File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.cpp
class Solution {
public:
    bool judgeSquareSum(int c) {
        long long a = 0;
        long long b = sqrt(c);

        while(a<=b){
            long long res = a*a+b*b;
            if(res==c) return true;
            else if(res<c) a++;
            else b--;
        }

        return false;
    }
};

File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.cs
using System;

public class Solution {
    public bool JudgeSquareSum(int c) {
        long a = 0;
        long b = (long)Math.Sqrt(c);

        while (a <= b) {
            long res = a * a + b * b;
            if (res == c) {
                return true;
            }
            else if (res < c) {
                a++;
            }
            else {
                b--;
            }
        }

        return false;
    }
}
